pkg/controller/import-controller.go

```diff
@@ -44,315 +51,532 @@ const (
+			"pvc.annotations.endpoint", checkPVC(pvc, AnnEndpoint), "pvc.annotations.source", checkPVC(pvc, AnnSource))
+		return reconcile.Result{}, nil
 	}
-	anno := map[string]string{}
 
 	// In case this is a request to create a blank disk on a block device, we do not create a pod.
 	// we just mark the DV as successful
 	volumeMode := getVolumeMode(pvc)
-	if volumeMode == v1.PersistentVolumeBlock && pvc.GetAnnotations()[AnnSource] == SourceNone {
-		klog.V(3).Infof("attempting to create blank disk for block mode, this is a no-op, marking pvc with pod-phase succeeded")
-		anno[AnnPodPhase] = string(v1.PodSucceeded)
-		_, err := updatePVC(ic.clientset, pvc, anno, lab)
-		if err != nil {
-			return errors.WithMessage(err, "could not update pvc %q annotation and/or label")
+	if volumeMode == corev1.PersistentVolumeBlock && pvc.GetAnnotations()[AnnSource] == SourceNone {
+		log.V(1).Info("attempting to create blank disk for block mode, this is a no-op, marking pvc with pod-phase succeeded")
+		if pvc.GetAnnotations() == nil {
+			pvc.SetAnnotations(make(map[string]string, 0))
 		}
-		return nil
-	}
-	// find import Pod
-	pod, err := ic.findImportPodFromCache(pvc)
-	if err != nil {
-		return err
+		pvc.GetAnnotations()[AnnPodPhase] = string(corev1.PodSucceeded)
+		if err := r.updatePVC(pvc); err != nil {
+			return reconcile.Result{}, errors.WithMessage(err, fmt.Sprintf("could not update pvc %q annotation and/or label", pvc.Name))
+		}
+		return reconcile.Result{}, nil
 	}
-	pvcKey, err := cache.MetaNamespaceKeyFunc(pvc)
+
+	// See if we have a pod associated with the PVC, we know the PVC has the needed annotations.
+	pod, err := r.findImporterPod(pvc)
 	if err != nil {
-		return err
+		return reconcile.Result{}, err
 	}
 
-	// Pod must be controlled by this PVC
-	if pod != nil && !metav1.IsControlledBy(pod, pvc) {
-		return errors.Errorf("found pod %s/%s not owned by pvc %s/%s", pod.Namespace, pod.Name, pvc.Namespace, pvc.Name)
-	}
+	return r.reconcilePvc(pvc, pod)
+}
 
-	// expectations prevent us from creating multiple pods. An expectation forces
-	// us to observe a pod's creation in the cache.
-	needsSync := ic.podExpectations.SatisfiedExpectations(pvcKey)
+func (r *ImportReconciler) findImporterPod(pvc *corev1.PersistentVolumeClaim) (*corev1.Pod, error) {
+	podName := fmt.Sprintf("%s-%s", common.ImporterPodName, pvc.Name)
+	pod := &corev1.Pod{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: podName, Namespace: pvc.GetNamespace()}, pod)
 
-	// make sure not to reprocess a PVC that has already completed successfully,
-	// even if the pod no longer exists
-	previousPhase, exists := pvc.ObjectMeta.Annotations[AnnPodPhase]
-	if exists && (previousPhase == string(v1.PodSucceeded)) {
-		needsSync = false
+	if k8serrors.IsNotFound(err) {
+		return nil, nil
 	}
 
-	if pod == nil && needsSync {
-		return ic.createImporterPod(pvc, pvcKey)
+	// Check if the PVC owns the pod.
+	pvcOwns := false
```

Comment:
maybe call `metav1.IsControlledBy` here?

Reply:
Was copying what was there, but this is better.